middleware/gatekeeper.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response, NextFunction } from 'express';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import rateLimiter from './rateLimiter';
+
+const gatekeeper = (roles: string[] = []) => {
```

Comment:
great work now how can we make it as in general for each api integrated to our server?

Reply:
through generic function ? ====> template